processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -191,11 +211,91 @@ private ColumnAnalysis analyzeStringColumn(
       }
     }
 
+    if (analyzingMinMax() && cardinality > 0) {
+      min = Strings.nullToEmpty(bitmapIndex.getValue(0));
+      max = Strings.nullToEmpty(bitmapIndex.getValue(cardinality - 1));
+    }
+
     return new ColumnAnalysis(
         capabilities.getType().name(),
         capabilities.hasMultipleValues(),
         size,
         analyzingCardinality() ? cardinality : 0,
+        min,
+        max,
+        null
+    );
+  }
+
+  private ColumnAnalysis analyzeStringColumn(
+      final ColumnCapabilities capabilities,
+      final StorageAdapter storageAdapter,
+      final String columnName
+  )
+  {
+    int cardinality = 0;
+    long size = 0;
+
+    Comparable min = null;
+    Comparable max = null;
+
+    if (analyzingCardinality()) {
+      cardinality = storageAdapter.getDimensionCardinality(columnName);
+    }
+
+    if (analyzingSize()) {
```

Comment:
Just for discussion, I think this PR Is fine:

How useful is the current method of size calculation? 

If we keep the "size" functionality, is the current "flat file" size estimate be more or less useful than returning the actual on-disk size of the segments (or expected size, for IncrementalIndexes)?